code/modules/mob/living/carbon/carbon.dm

```diff
@@ -251,17 +251,22 @@
 					"<span class='notice'>You pat [src] on the head.</span>",\
 					)
 				else
-
-					M.visible_message(\
-					"<span class='notice'>[M] gives [src] a [pick("hug","warm embrace")].</span>",\
-					"<span class='notice'>You hug [src].</span>",\
-					)
-					if(ishuman(src))
-						var/mob/living/carbon/human/H = src
-						if(H.wear_suit)
-							H.wear_suit.add_fingerprint(M)
-						else if(H.w_uniform)
-							H.w_uniform.add_fingerprint(M)
+					if(M.zone_selected == ("r_arm" || "r_hand" || "l_arm" || "l_hand"))
```

Comment:
This whole thing can be made a little neater if you move the `if()` onto the same line as the `else` before it, and indent the rest of the code accordingly. Like this:
```dm
if(M.zone_selected == "head")
	M.visible_message(\
	"<span class='notice'>[M] pats [src] on the head.</span>",\
	"<span class='notice'>You pat [src] on the head.</span>",\
	)
else if(M.zone_selected in list("r_arm", "r_hand", "l_arm", "l_hand"))
	M.visible_message(\
	"<span class='notice'>[M] shakes [src]'s hand.</span>",\
	"<span class='notice'>You shake hands with [src].</span>",\
	)
else
	M.visible_message(\
	"<span class='notice'>[M] gives [src] a [pick("hug","warm embrace")].</span>",\
	"<span class='notice'>[M] You hug [src].</span>",\
	)
```